src/stirling/source_connectors/socket_tracer/conn_tracker.cc

```diff
@@ -962,7 +982,9 @@ void ConnTracker::InferConnInfo(system::ProcParser* proc_parser,
     bool success = conn_resolver_->Setup();
     if (!success) {
       conn_resolver_.reset();
-      conn_resolution_failed_ = true;
+      if (!parse_local_addr_only) {
+        conn_resolution_failed_ = true;
+      }
```

Comment:
Why is this conditional on `!parse_local_addr_only`?

Reply:
We hit this condition when the `FDResolver` fails to [read](https://github.com/pixie-io/pixie/blob/07672ff59b9441242a7adf8993a6493572b457ac/src/stirling/source_connectors/socket_tracer/fd_resolver.cc#L39) the `proc/pid/fd` link of the connection. If we set `conn_resolution_failed_` for cases where we already know the remote IP, then we'll end up [disabling](https://github.com/pixie-io/pixie/blob/6b418c2d4a3b69e8b37b76b6b4ff38bcde80fb8b/src/stirling/source_connectors/socket_tracer/conn_tracker.cc#L729) [tracing](https://github.com/pixie-io/pixie/blob/6b418c2d4a3b69e8b37b76b6b4ff38bcde80fb8b/src/stirling/source_connectors/socket_tracer/conn_tracker.cc#L754) in `ConnTracker::UpdateState` and skipping inference.

That said, I’m unsure why the FD might be volatile for connections where we know the remote addr already.

Before I added this conditional, a couple bpf tests started failing on [conn_resolver_->Setup()](https://github.com/pixie-io/pixie/blob/6b418c2d4a3b69e8b37b76b6b4ff38bcde80fb8b/src/stirling/source_connectors/socket_tracer/conn_tracker.cc#L982).
```sh
pgsql_trace_bpf_test:PostgreSQLTraceTest.SelectQuery
socket_trace_bpf_test:NonVecSyscalls/NonVecSyscallTests.NonVecSyscalls/1
```
See [buildbuddy](https://app.buildbuddy.io/invocation/2c6c36ad-ae6d-4d34-a7d7-d4fc07e36a2e?target=%2F%2Fsrc%2Fstirling%2Fsource_connectors%2Fsocket_tracer%3Apgsql_trace_bpf_test&targetStatus=6#@214) for additional context.